### PR TITLE
Feat/swarinfo 294 broadcast communication

### PR DESCRIPTION
**Pull request by g-lesssard**
Ros implementation for broadcast. Took the broker implementation from swarmus-ros with some changes to add it in the project to simplify uniformity.

### CODE REVIEW

File: src/network/src/ros/src/broker/CommunicationBroker.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+std::vector<uint16_t> CommunicationBroker::getRobotList() {
+    std::vector<uint16_t> robotList;
+    ros::NodeHandle nh("~");
+
+    XmlRpc::XmlRpcValue configRobotList;
+    std::string robotListParamName =
+        nh.param(std::string("configList"), std::string("/Broker/robots"));
+    if (!nh.getParam("/Broker/robots", configRobotList)) {
```
Comment (xgroleau) [inline src/network/src/ros/src/broker/CommunicationBroker.cpp:25]: What we could do instead of hardcoding is at boot, the broadcast or unicast sends a ros message on a topic with their ID, and the communication broker catches them and stores thoses id. though that could go in another PR, the file is fine for now.

**Review by xgroleau — CHANGES_REQUESTED** [review]
Very minor comments, GJ honestly 👍

File: src/main.cpp [hunk 2]
```diff
@@ -140,6 +141,67 @@ class UnicastMessageDispatcher : public AbstractTask<3 * configMINIMAL_STACK_SIZ
...
+        MessageDispatcher dispatcher =
+            MessageHandlerContainer::createMessageDispatcher(deserializer, networkApiHandler);
+
+        while (NetworkContainer::getNetworkManager().getNetworkStatus() !=
+               NetworkStatus::Connected) {
+            Task::delay(500);
+        }
+        while (true) {
+            if (!dispatcher.deserializeAndDispatch()) {
+                m_logger.log(LogLevel::Error, "Fail to deserialize/dispatch unicast");
```
[the remark this answers is not in the document]
Comment (g-lesssard) [reply src/main.cpp:199]: Probably forgot to edit after copy-paste

File: src/network/src/ros/src/NetworkContainer.cpp [hunk 3]
```diff
@@ -29,4 +31,13 @@ INetworkOutputStream& NetworkContainer::getNetworkOutputStream() {
     static NetworkOutputStream s_outputStream(LoggerContainer::getLogger());
 
     return s_outputStream;
+}
+
+INetworkBroadcast& NetworkContainer::getNetworkBroadcast() {
+    // Could be supplied by launch file when integrated in simulation
+    std::string outputPrefix("/Communication/broadcastOutput/");
+    std::string inputPrefix("/Communication/broadcastOutput/");
```
[the remark this answers is not in the document]
Comment (g-lesssard) [reply src/network/src/ros/src/NetworkContainer.cpp]: Yeah, I'll add a define to ensure consistency with the broker and tests

File: src/network/src/ros/src/broker/CommunicationBroker.cpp (hunk 1 above)
Comment (g-lesssard) [reply src/network/src/ros/src/broker/CommunicationBroker.cpp:25]: I like this idea, we could also use a similar mechanism if we spawn and close dynamically for some reason

**Review by xgroleau — APPROVED** [review]
